src/plopp/prep.py

```diff
@@ -27,7 +27,16 @@ def _convert_if_not_none(x, unit):
     return x
 
 
-def preprocess(obj, crop=None, name=''):
+def _check_input_data_size(obj):
+    limits = {1: 1_000_000, 2: 5000 * 5000}
```

Comment:
Regarding your question, I think we should determine limits based on approximate timings of creating such a plot.

Reply:
I think there is both figure creation and then interactions like zooming which can also be slow. I think it's quite hard to determine but we should experiment?

Reply:
I get the following timings on my workstation:

**2d image:**
- 10000x10000: creation=1min, zoom=20s
- 5000x5000: creation=14s, zoom=5s

**1d marker plot:**
- 1M points: creation=3s, zoom=<1s
- 5M points: creation=14s, zoom=1s
- 10M points: creation=28s, zoom=2s

**1d line plot (no markers):**
- 5M points: creation=2s, zoom=<1s

So should we go with 5M points, and 5000x5000?

Reply:
I think we should choose the limits such that wait time does not exceed a couple of seconds. 10 seconds is already too long probably.

Reply:
So i'll go with 1M and 2500x2500